### PR TITLE
Add Black Chocobo

### PR DESCRIPTION
**Pull request by sithlord48**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.     Link:  www.github.com/sithlord48/blackchocobo
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* 
  Stable version needs the patches,  future releases have them [MR#57](https://github.com/sithlord48/blackchocobo/pull/57)  & [MR#58](https://github.com/sithlord48/blackchocobo/pull/58)

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.sithlord48.BlackChocobo.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+app-id: org.sithlord48.BlackChocobo
```
Comment (hfiguiere) [inline org.sithlord48.BlackChocobo.yml:1]: `sithlord48.org` doesn't exist. the app-id doesn't meet the requirements.

File: org.sithlord48.BlackChocobo.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+default-branch: stable
```
Comment (hfiguiere) [inline org.sithlord48.BlackChocobo.yml:2]:
```suggestion
```

File: org.sithlord48.BlackChocobo.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+runtime: org.kde.Platform
+runtime: org.kde.Sdk.Locale
```
Comment (hfiguiere) [inline org.sithlord48.BlackChocobo.yml:4]:
```suggestion
```

File: org.sithlord48.BlackChocobo.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+sdk: org.kde.Sdk
+runtime-version: 6.4
+command: blackchocobo
+finish-args:
+  - --filesystem=host
```
Comment (hfiguiere) [inline org.sithlord48.BlackChocobo.yml:9]: that's a very broad permission

File: org.sithlord48.BlackChocobo.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline org.sithlord48.BlackChocobo.yml:11]: missing `--share=ipc`

File: org.sithlord48.BlackChocobo.yml (hunk 4 above)
Comment (sithlord48) [reply org.sithlord48.BlackChocobo.yml:9]: Is there a more limited way to allow r/w of user files

Comment (sithlord48) [reply org.sithlord48.BlackChocobo.yml:9]: Changed to home

File: com.github.sithlord48.blackchocobo.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+app-id: com.github.sithlord48.blackchocobo
```
Comment (hfiguiere) [inline com.github.sithlord48.blackchocobo.yml:1]: it should be `io.github` as per the documentation.

File: com.github.sithlord48.blackchocobo.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+runtime: org.kde.Platform
+sdk: org.kde.Sdk
+runtime-version: '6.4'
+command: blackchocobo
+finish-args:
+  - --filesystem=home
+  - --filesystem=xdg-data
```
Comment (hfiguiere) [inline com.github.sithlord48.blackchocobo.yml:8]:
```suggestion
```
no.

File: org.sithlord48.BlackChocobo.yml (hunk 1 above)
Comment (sithlord48) [reply org.sithlord48.BlackChocobo.yml]: will use `com.github.sithlord48.blackchocobo`

**Comment by sithlord48** [comment]
I have to use `--filesystem=host` 

There is no way to override the portal dialog(s) to allow a save name to be suggested when the current save as type changes. Todo this I am forced to use the Qt's Non Native Dialog. I'm not talking about just adding ".ext" to a file either just about every format that is save has some rules around what file names are valid so for each type a suggestion is provided. For example the psv format requires the output filename to be parts of the file contents in the form of shift-jis hex char values. Its just not reasonable to expect my users to know these things[[1]](https://github.com/sithlord48/blackchocobo/wiki/PSX-Formats#psv-format). That is one reason they are using this app and not hexediting by hand, While There is an option in the application to use force the use of the os' native dialog when doing so users do lose all name suggestions. For this reason I had to switch back using host its the only way I can give users both suggestions and the ability to save / load from mounted usb / network drives . This app works with console saves so users often are reading from a mounted device outside of their home folder.

**Comment by barthalion** [comment]
bot, build io.github.sithlord48.blackchocobo

**Comment by barthalion** [comment]
f-e-d-c stanza fails validation

**Comment by sithlord48** [comment]
> f-e-d-c stanza fails validation

Fixed

**Comment by barthalion** [comment]
bot, build io.github.sithlord48.blackchocobo

**Comment by sithlord48** [comment]
Use correct name `metainfo`  not `metadata`

**Comment by barthalion** [comment]
bot, build io.github.sithlord48.blackchocobo

**Comment by sithlord48** [comment]
Fixed: / instead of . for path

**Comment by sithlord48** [comment]
bot, build io.github.sithlord48.blackchocobo

**Comment by sithlord48** [comment]
bot, build io.github.sithlord48.blackchocobo

**Comment by sithlord48** [comment]
bot, build io.github.sithlord48.blackchocobo

**Pull request closed without merging**